JAVA1/src/Solution/FileSearchInHomeDirectory.java

```diff
@@ -0,0 +1,54 @@
+package Solution;
+import java.io.File;
+import java.util.*;
+import java.util.regex.*;
+public class FileSearchInHomeDirectory {
```

Comment:
FileSearchInHomeDirecory means you are restricting the search to home directory. But your code allows flexible search. Rename a file to more suitable name that clearly talk about its functionality.